build.gradle

```diff
@@ -0,0 +1,78 @@
+    id 'org.jetbrains.kotlin.jvm' version '1.3.72'
+}
+
+group = 'com.cag.cagbackendapi'
+version = '0.0.1-SNAPSHOT'
+sourceCompatibility = '11'
+
+repositories {
+	mavenCentral()
+}
+
+dependencies {
+	implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8"
+	implementation "org.apache.tomcat.embed:tomcat-embed-core:9.0.35"
+	implementation "org.springframework.boot:spring-boot-starter-data-jpa"
+	implementation "com.nimbusds:nimbus-jose-jwt:8.2.1"
+	implementation 'org.springframework.boot:spring-boot-starter-web'
+	implementation "org.springframework.retry:spring-retry:1.2.4.RELEASE"
+	implementation "org.springframework.boot:spring-boot-starter-aop:1.5.22.RELEASE"
+	implementation "org.springframework.boot:spring-boot-starter-actuator"
+	implementation "org.springframework.boot:spring-boot-starter-data-rest"
+	implementation 'junit:junit:4.12'
+	implementation "org.springframework:spring-web:5.2.3.RELEASE"
+	implementation "org.dom4j:dom4j:2.1.3"
+	implementation "com.fasterxml.jackson.module:jackson-module-kotlin:2.10.0"
+	implementation "org.jetbrains.kotlin:kotlin-reflect:1.3.50"
+	implementation "com.google.code.gson:gson:2.8.5"
+	//implementation "org.flywaydb:flyway-core"
+	implementation "io.micrometer:micrometer-registry-prometheus"
+	implementation "com.fasterxml.jackson.core:jackson-core:2.10.0"
+	implementation "org.springdoc:springdoc-openapi-ui:1.2.24"
+	implementation "io.swagger:swagger-annotations:1.5.0"
+	implementation "com.fasterxml.jackson.core:jackson-databind:2.10.0"
+
+	runtimeOnly "org.postgresql:postgresql:42.2.13"
+
+	testImplementation 'org.springframework.kafka:spring-kafka-test'
+	testImplementation 'org.junit.platform:junit-platform-runner:1.2.0'
+	testImplementation 'org.mockito:mockito-junit-jupiter:3.0.0'
+	testImplementation 'com.nhaarman:mockito-kotlin:1.5.0'
+	testImplementation 'org.mockito:mockito-core:3.0.0'
+	testImplementation "org.springframework.boot:spring-boot-starter-test"
+	testImplementation "org.springframework.security:spring-security-test"
+	testImplementation "org.springframework.boot:spring-boot-starter-test:2.1.7.RELEASE"
+	testImplementation "org.junit.jupiter:junit-jupiter-api:5.5.1"
+	testImplementation "org.junit.platform:junit-platform-runner:1.5.1"
+	testImplementation "org.junit.platform:junit-platform-commons:1.5.1"
+	testImplementation "org.mockito:mockito-junit-jupiter:3.2.4"
+	testImplementation "com.nhaarman.mockitokotlin2:mockito-kotlin:2.2.0"
+	testImplementation "org.mockito:mockito-core:3.3.0"
+	testImplementation('org.springframework.boot:spring-boot-starter-test') {
+		exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
+	}
+
+	testRuntimeOnly "org.junit.jupiter:junit-jupiter-engine:5.4.0"
+	testRuntimeOnly "org.junit.platform:junit-platform-launcher:1.4.2"
+	testRuntimeOnly "org.junit.vintage:junit-vintage-engine:5.4.0"
+}
+
+test {
+	useJUnitPlatform()
+}
+
+compileKotlin {
+    kotlinOptions {
+        jvmTarget = "1.8"
```

Comment:
What does 1.8 mean in this context?

Reply:
It means that kotlin will read the java bytecode as jdk 1.8 (Java 8). In this project we'll be using Java 11, but Kotlin will read the java 11 bytecode as java 8 bytecode - which is fine because the java 11 bytecode is the same as java 8 bytecode from the perspective of kotlin. Kotlin views it as the same. In a future kotlin update, we may change our jvmTarget to be "11" instead of "1.8".